sysutils/osxbom/Portfile

```
@@ -0,0 +1,68 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem          1.0
+PortGroup           github 1.0
+
+# My repo is ahead of the main one in terms of versions:
+github.setup        cooljeanius osxbom 0.0.5
+github.tarball_from archive
+categories          sysutils
+maintainers         {gwmail.gwu.edu:egall @cooljeanius} openmaintainer
+platforms           darwin
+license             public-domain
+
+description         A free re-implementation of the NextSTEP/OSX lsbom \
+                    utility
+long_description    ${description}. Used by the PureDarwin project \
+                    to work with MacPorts binary packages.
+homepage            https://cauldrondevelopment.com/svn/${name}/trunk/
+
+checksums           rmd160  ae9fa1120123aa0f19d1a92bce869d530556bf19 \
+                    sha256  788c6650494eff481baebe1129335003301d9530c20ad61cb0ec013b628797ca \
+                    size    155958
+
+# probably not actually necessary to autoreconf, but whatever:
+use_autoreconf      yes
+autoreconf.args     -fvi -Wall
+
+# mostly just things picked up by trace mode:
+depends_build-append \
+                    port:gawk \
+                    port:grep \
+                    bin:nroff:groff \
+                    bin:gmkdir:coreutils \
+                    port:libmacho-headers
+depends_lib-append  port:libunwind
+
+set CFLAGS          "${configure.cflags} [get_canonical_archflags cc]"
+
+build.env-append    CC=${configure.cc} \
+                    CXX=${configure.cxx} \
+                    CPP=${configure.cpp}
+
+pre-build {
+    reinplace "s|-Werror|${CFLAGS}|" ${worksrcpath}/Makefile
+}
+
+test.run            yes
+test.target         check
+test.target-append  distcheck
+depends_test-append bin:gzip:gzip \
+                    bin:zip:zip
+
+destroot {
+        xinstall ${worksrcpath}/lsbom ${destroot}${prefix}/bin
+        xinstall ${worksrcpath}/bom.h ${destroot}${prefix}/include
+}
+
+variant rename description {Stop osxbom from shadowing the system lsbom} {
+    destroot {
+        xinstall ${worksrcpath}/lsbom ${destroot}${prefix}/bin/${name}
+        xinstall -d ${destroot}${prefix}/include/${name}
+        xinstall ${worksrcpath}/bom.h ${destroot}${prefix}/include/${name}
+    }
+}
+
+default_variants    +rename
```

IMHO, this project should provide its only binary as `osxbom` to prevent shadowing the system's `lsbom` (without the need of a variant), and to provide consistency across all macOS versions supported by MacPorts.

```suggestion
destroot {
    xinstall -d ${destroot}${prefix}/include
    xinstall -m 0755 ${worksrcpath}/lsbom ${destroot}${prefix}/bin/${name}
    xinstall -m 0644 ${worksrcpath}/bom.h ${destroot}${prefix}/include
}
```

---

The idea is for this port to be available on systems like PureDarwin that don't come with their own `lsbom` though.

---

I think this can be achieved through a variant that users can activate rather than it being activated by default. 

```tcl
variant lsbom description {Overwrite the system's lsbom with osxbom} {
    post-destroot {
        ln -s ${name} ${destroot}${prefix}/bin/lsbom
    }
}
```

---

That's basically what I had originally in the Trac ticket where I started this, except I'd started the variant name with the `+no_` phrasing, which @ryandesign said to invert, which is how I arrived at the current setup instead.

---

The variant name I suggested depicts the function of the variant, even in the possible command to activate it (e.g `sudo port install osxbom +lsbom`, so I think Ryan would be fine with it.

---

ok, switched to doing the variant that way in 5cb5b8c.